### PR TITLE
Added infrastructure for reload of messages after disconnect

### PR DESCRIPTION
**Pull request by hnrkndrssn**
The result after calling the new event for displaying a reload messages notification
ReloadMessage

### CODE REVIEW

File: JabbR/Chat.js [hunk 1]
```diff
@@ -909,6 +909,10 @@
             });
     });
 
+    $(ui).bind(ui.events.reloadMessages, function (ev, roomInfo) {
+        ui.showReloadMessage(roomInfo.name);
```
Comment (davidfowl) [inline JabbR/Chat.js]: What is this?

Comment (davidfowl) [inline JabbR/Chat.js]: Who triggers this?

Comment (hnrkndrssn) [inline JabbR/Chat.js]: That would be called when it is detected that a room is missing messages due to a disconnection (for example after resuming computer from sleep).

Or have I assumed incorrectly?

Sent from my Windows Phone

**Comment by davidfowl** [comment]
Shouldn't that get triggered when clicking on the reload button?

**Comment by hnrkndrssn** [comment]
Oh, sorry I misunderstood then. That event is just for displaying that notification.

I didn't do the event for actually reloading the messages. I can add that tonight.

Sent from my Windows Phone

**Comment by hnrkndrssn** [comment]
And I will work on the naming of the events as well...:|

Sent from my Windows Phone

File: JabbR/Chat.ui.js [hunk 2]
```diff
@@ -1910,6 +1929,9 @@
             },
             updateTimeout);
         },
+        reloadMessages: function(roomName) {
```
Comment (davidfowl) [inline JabbR/Chat.ui.js]: Remove this.

File: JabbR/Chat.js [hunk 3]
```diff
@@ -909,6 +909,10 @@
             });
     });
 
+    $(ui).bind(ui.events.reloadMessages, function (ev, roomInfo) {
+        ui.reloadMessages(roomInfo);
```
Comment (davidfowl) [inline JabbR/Chat.js]: Remove this.

File: JabbR/Chat.ui.js [hunk 4]
```diff
@@ -784,6 +785,16 @@
         }, cycleTimeInMilliseconds);
     }
 
+    function showReloadMessageNotification(roomName) {
+        var room = getRoomElements(roomName);
+        if (!room.isLobby()) {
```
Comment (davidfowl) [inline JabbR/Chat.ui.js]: What happens if detection happens and I'm on the lobby? It should always show up. I'm not sure I want to read messages per room.

Comment (hnrkndrssn) [inline JabbR/Chat.ui.js]: OK, so you want the reload to reload all messages in all open rooms? In that case I'll remove the room name params

File: JabbR/Chat.ui.js [hunk 5]
```diff
@@ -1910,6 +1917,10 @@
             },
             updateTimeout);
         },
+        showReloadMessageNotification: function () {
+            $reloadMessageNotification.appendTo($chatArea);
```
Comment (davidfowl) [inline JabbR/Chat.ui.js:1921]: If I call this mutliple times does it append multiple elements? Should there be a hide?

Comment (hnrkndrssn) [inline JabbR/Chat.ui.js:1921]: I have tested calling this multiple times from different areas in the code and it only ever adds the one element.

**Pull request merged**